src/nvim/os_unix.c

```diff
@@ -186,6 +186,7 @@ void mch_exit(int r)
   free_all_mem();
 #endif
 
+  stream_set_blocking(global_input_fd, true);  //normalize stream (#2598)
```

Comment:
If I move this above `event_teardown()` (which stops `uv_default_loop`), the result is:

    nvim: /home/vagrant/neovim2/.deps/build/src/libuv/src/unix/core.c:833: uv__io_stop: Assertion `loop->watchers[w->fd] == w' failed.